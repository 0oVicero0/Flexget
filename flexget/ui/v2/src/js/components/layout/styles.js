import styled from 'styled-components';
import theme from 'theme';

const HEADER_HEIGHT = 5;
const MOBILE_HEADER_HEIGHT = (HEADER_HEIGHT * 2) - 0.2;
const PADDING = 1;

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  min-height: 100vh;
`;

export const Main = styled.main`
  display: flex;
  flex-direction: row;
  margin-top: ${MOBILE_HEADER_HEIGHT}rem;
  flex: 1;
  ${theme.breakpoints.up('sm')} {
    margin-top: ${HEADER_HEIGHT}rem;
  }
`;

export const Header = styled.header`
  display: flex;
  min-height: ${HEADER_HEIGHT}rem;
  flex-direction: column;
  z-index: 2;
  position: fixed;
  width: 100%;
  ${theme.breakpoints.up('sm')} {
    flex-direction: row;
  }
`;

export const LogoWrapper = styled.div`
  height: ${HEADER_HEIGHT}rem;
`;

export const Nav = styled.nav`
  height: ${HEADER_HEIGHT}rem;
  flex: 1;
`;

export const SideBar = styled.aside`
  overflow-y: auto;
`;

export const Content = styled.section`
  flex: 1;
  padding: ${PADDING}rem;
  overflow-y: auto;
  opacity: 1;
  transition: ${theme.transitions.create(['opacity', 'margin-left'])};
  margin-left: ${({ open }) => (open ? '19rem' : '5rem')};

  ${theme.breakpoints.down('sm')} {
    margin-left: 0;
    padding: ${({ open }) => (open ? 0 : `${PADDING}rem`)};
    opacity: ${({ open }) => (open ? 0 : 1)};
    display: ${({ open }) => (open ? 'none' : 'block')};
  }
`;
