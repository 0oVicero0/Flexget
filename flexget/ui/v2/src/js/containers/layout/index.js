import { connect } from 'react-redux';
import { checkLogin } from 'actions/auth';
import Layout from 'components/layout';

export function mapStateToProps({ auth }) {
  return {
    loggedIn: !!auth.loggedIn,
  };
}

export function mapDispatchToProps(dispatch) {
  return {
    checkLogin: () => dispatch(checkLogin()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Layout);
