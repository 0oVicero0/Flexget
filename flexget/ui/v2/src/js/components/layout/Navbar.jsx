import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { withStyles, createStyleSheet } from 'material-ui/styles';
import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import IconButton from 'material-ui/IconButton';
import Icon from 'material-ui/Icon';
import Menu, { MenuItem } from 'material-ui/Menu';
import 'font-awesome/css/font-awesome.css';

const styleSheet = createStyleSheet('Navbar', theme => ({
  appBar: {
    position: 'static',
  },
  spacer: {
    flex: 1,
  },
  toolbar: {
    backgroundColor: theme.palette.primary[800],
    height: '100%',
  },
  icon: {
    color: theme.palette.getContrastText(theme.palette.primary[800]),
  },
  menuIcon: {
    paddingRight: 30,
  },
}));

class Navbar extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    toggle: PropTypes.func.isRequired,
    logout: PropTypes.func.isRequired,
  };

  state = {
    open: false,
  };

  handleRequestClose = () => this.setState({ open: false })

  handleMenuClick = event => this.setState({
    open: true,
    anchorEl: event.currentTarget,
  })

  render() {
    const { classes, toggle, logout } = this.props;
    const { anchorEl, open } = this.state;

    return (
      <AppBar className={classes.appBar}>
        <Toolbar className={classes.toolbar}>
          <IconButton
            className={classes.icon}
            onClick={toggle}
          >
            <Icon className="fa fa-bars" />
          </IconButton>
          <div className={classes.spacer} />
          <Link to="/config">
            <IconButton
              className={classes.icon}
              aria-label="Config"
            >
              <Icon className="fa fa-pencil" />
            </IconButton>
          </Link>
          <Link to="/log">
            <IconButton
              className={classes.icon}
              aria-label="Log"
            >
              <Icon className="fa fa-file-text-o" />
            </IconButton>
          </Link>
          <IconButton
            className={classes.icon}
            aria-label="Manage"
            onClick={this.handleMenuClick}
          >
            <Icon className="fa fa-cog" />
          </IconButton>
          <Menu
            id="nav-menu"
            anchorEl={anchorEl}
            open={open}
            onRequestClose={this.handleRequestClose}
          >
            <MenuItem>
              <Icon className={`${classes.menuIcon} fa fa-refresh`} />
              Reload
            </MenuItem>
            <MenuItem>
              <Icon className={`${classes.menuIcon} fa fa-power-off`} />
              Shutdown
            </MenuItem>
            <MenuItem>
              <Icon className={`${classes.menuIcon} fa fa-database`} />
              Database
            </MenuItem>
            <MenuItem onClick={logout}>
              <Icon className={`${classes.menuIcon} fa fa-sign-out`} />
              Logout
            </MenuItem>
          </Menu>
        </Toolbar>
      </AppBar>
    );
  }
}

export default withStyles(styleSheet)(Navbar);
