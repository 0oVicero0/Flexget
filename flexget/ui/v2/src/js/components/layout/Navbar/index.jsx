import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { withStyles, createStyleSheet } from 'material-ui/styles';
import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import IconButton from 'material-ui/IconButton';
import Icon from 'material-ui/Icon';
import Menu, { MenuItem } from 'material-ui/Menu';
import Dialog, {
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
} from 'material-ui/Dialog';
import Button from 'material-ui/Button';
import 'font-awesome/css/font-awesome.css';

const styleSheet = createStyleSheet('Navbar', theme => ({
  appBar: {
    position: 'static',
  },
  spacer: {
    flex: 1,
  },
  toolbar: {
    backgroundColor: theme.palette.primary[800],
    height: '100%',
  },
  icon: {
    color: theme.palette.getContrastText(theme.palette.primary[800]),
  },
  menuIcon: {
    paddingRight: 30,
  },
}));

export class Navbar extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    toggle: PropTypes.func.isRequired,
    logout: PropTypes.func.isRequired,
    reloadServer: PropTypes.func.isRequired,
    shutdownServer: PropTypes.func.isRequired,
  };

  state = {
    menuOpen: false,
    shutdownPrompt: false,
  };

  handleMenuRequestClose = () => this.setState({
    menuOpen: false,
    anchorEl: undefined,
  })

  handleMenuClick = event => this.setState({
    menuOpen: true,
    anchorEl: event.currentTarget,
  })

  handleReloadClick = () => {
    this.props.reloadServer();
    this.handleMenuRequestClose();
  }

  handleShutdownPromptClick = () => {
    this.setState({ shutdownPrompt: true });
    this.handleMenuRequestClose();
  }

  handleShutdownClick = () => {
    this.props.shutdownServer();
    this.handleShutdownRequestClose();
  }

  handleShutdownRequestClose = () => this.setState({
    shutdownPrompt: false,
  });

  render() {
    const { classes, toggle, logout } = this.props;
    const { anchorEl, menuOpen, shutdownPrompt } = this.state;

    return (
      <AppBar className={classes.appBar}>
        <Toolbar className={classes.toolbar}>
          <IconButton
            className={classes.icon}
            onClick={toggle}
          >
            <Icon className="fa fa-bars" />
          </IconButton>
          <div className={classes.spacer} />
          <Link to="/config">
            <IconButton
              className={classes.icon}
              aria-label="Config"
            >
              <Icon className="fa fa-pencil" />
            </IconButton>
          </Link>
          <Link to="/log">
            <IconButton
              className={classes.icon}
              aria-label="Log"
            >
              <Icon className="fa fa-file-text-o" />
            </IconButton>
          </Link>
          <IconButton
            className={classes.icon}
            aria-label="Manage"
            onClick={this.handleMenuClick}
          >
            <Icon className="fa fa-cog" />
          </IconButton>
          <Menu
            id="nav-menu"
            anchorEl={anchorEl}
            open={menuOpen}
            onRequestClose={this.handleMenuRequestClose}
          >
            <MenuItem onClick={this.handleReloadClick}>
              <Icon
                className={`${classes.menuIcon} fa fa-refresh`}
              />
              Reload
            </MenuItem>
            <MenuItem onClick={this.handleShutdownPromptClick}>
              <Icon
                className={`${classes.menuIcon} fa fa-power-off`}
              />
              Shutdown
            </MenuItem>
            <MenuItem>
              <Icon className={`${classes.menuIcon} fa fa-database`} />
              Database
            </MenuItem>
            <MenuItem onClick={logout}>
              <Icon className={`${classes.menuIcon} fa fa-sign-out`} />
              Logout
            </MenuItem>
          </Menu>
          <Dialog open={shutdownPrompt} onRequestClose={this.handleShutdownRequestClose}>
            <DialogTitle>Shutdown</DialogTitle>
            <DialogContent>
              <DialogContentText>
                Are you sure you want to shutdown FlexGet?
              </DialogContentText>
            </DialogContent>
            <DialogActions>
              <Button onClick={this.handleShutdownRequestClose} color="primary">
                Cancel
              </Button>
              <Button onClick={this.handleShutdownClick} color="primary">
                Shutdown
              </Button>
            </DialogActions>
          </Dialog>
        </Toolbar>
      </AppBar>
    );
  }
}

export default withStyles(styleSheet)(Navbar);
