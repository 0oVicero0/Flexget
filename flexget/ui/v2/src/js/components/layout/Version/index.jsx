import React, { Component } from 'react';
import PropTypes from 'prop-types';
import semver from 'semver-compare';
import IconButton from 'material-ui/IconButton';
import Icon from 'material-ui/Icon';
import { withStyles, createStyleSheet } from 'material-ui/styles';
import 'font-awesome/css/font-awesome.css';

const styleSheet = createStyleSheet('Version', theme => ({
  version: {
    color: theme.palette.error[500],
  },
  par: {
    margin: 0,
  },
}));

class Version extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    version: PropTypes.shape({
      api: PropTypes.string,
      flexget: PropTypes.string,
      latest: PropTypes.string,
    }).isRequired,
    getVersion: PropTypes.func.isRequired,
  };

  componentDidMount() {
    this.props.getVersion();
  }

  render() {
    const { classes, version: { api, flexget, latest } } = this.props;
    return (
      <div className={classes.version}>
        <p className={classes.par}>Version Info</p>
        <p className={classes.par}>Flexget: { flexget } {
          latest && semver(latest, flexget) === 1 && (
            <IconButton href="https://flexget.com/ChangeLog">
              <Icon className={`fa fa-question-circle-o ${classes.version}`} />
            </IconButton>
          ) } </p>
        <p className={classes.par}>API: { api }</p>
      </div>
    );
  }
}

export default withStyles(styleSheet)(Version);
