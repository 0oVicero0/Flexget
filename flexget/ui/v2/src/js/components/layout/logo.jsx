import React from 'react';
import PropTypes from 'prop-types';
import headerImage from 'images/header.png';
import { withStyles, createStyleSheet } from 'material-ui/styles';

const styleSheet = createStyleSheet('Logo', theme => ({
  logo: {
    background: `${theme.palette.accent[900]} url(${headerImage}) no-repeat center`,
    backgroundSize: '175px',
    transition: 'width 0.5s ease',
    height: '100%',
    width: '100%',
  },
}));

const Logo = ({ classes }) => <div className={classes.logo} />;

Logo.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styleSheet)(Logo);
