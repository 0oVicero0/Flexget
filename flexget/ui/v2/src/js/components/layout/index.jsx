import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withStyles, createStyleSheet } from 'material-ui/styles';
import Logo from 'components/layout/Logo';
import Navbar from 'containers/layout/Navbar';
import SideNav from 'components/layout/Sidenav';
import LoadingBar from 'containers/common/LoadingBar';
import ErrorStatus from 'containers/common/ErrorStatus';
import InfoStatus from 'containers/common/InfoStatus';

const HEADER_HEIGHT = 50;
const MOBILE_HEADER_HEIGHT = (HEADER_HEIGHT * 2) - 2;
const PADDING = 10;

const styleSheet = createStyleSheet('Layout', theme => ({
  layout: {
    display: 'flex',
    flexDirection: 'column',
    height: '100%',
    minHeight: '100vh',
  },
  main: {
    display: 'flex',
    flexDirection: 'row',
    marginTop: MOBILE_HEADER_HEIGHT,
    flex: 1,
    [theme.breakpoints.up('sm')]: {
      marginTop: HEADER_HEIGHT,
    },
  },
  header: {
    display: 'flex',
    minHeight: HEADER_HEIGHT,
    flexDirection: 'column',
    zIndex: 2,
    position: 'fixed',
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      flexDirection: 'row',
    },
  },
  logo: {
    height: HEADER_HEIGHT,
  },
  navbar: {
    height: HEADER_HEIGHT,
    flex: 1,
  },
  sidebar: {
    overflowY: 'auto',
  },
  content: {
    flex: 1,
    padding: PADDING,
    overflowY: 'auto',
  },
}));

class Layout extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    classes: PropTypes.object.isRequired,
    loggedIn: PropTypes.bool,
    checkLogin: PropTypes.func.isRequired,
  };

  static defaultProps = {
    loggedIn: false,
  };

  state = {
    sideBarOpen: false,
  };

  componentDidMount() {
    const { loggedIn, checkLogin } = this.props;
    if (!loggedIn) {
      checkLogin();
    }
  }

  toggleSideBar = () => {
    this.setState({
      sideBarOpen: !this.state.sideBarOpen,
    });
  }

  render() {
    const { children, classes } = this.props;
    const { sideBarOpen } = this.state;

    return (
      <div className={classes.layout}>
        <header className={classes.header}>
          <div className={classes.logo}>
            <Logo sideBarOpen={sideBarOpen} />
          </div>
          <nav className={classes.navbar}>
            <Navbar toggle={this.toggleSideBar} />
            <LoadingBar />
          </nav>
        </header>
        <main className={classes.main}>
          <aside className={classes.sidebar}>
            <SideNav sideBarOpen={sideBarOpen} />
          </aside>
          <section className={classes.content}>
            { children }
          </section>
          <ErrorStatus />
          <InfoStatus />
        </main>
      </div>
    );
  }
}

export default withStyles(styleSheet)(Layout);
