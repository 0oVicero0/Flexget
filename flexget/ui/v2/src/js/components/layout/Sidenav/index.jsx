import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import classNames from 'classnames';
import { withStyles, createStyleSheet } from 'material-ui/styles';
import Button from 'material-ui/Button';
import List, { ListItem } from 'material-ui/List';
import Icon from 'material-ui/Icon';
import Paper from 'material-ui/Paper';
import Version from 'containers/layout/Version';
import 'font-awesome/css/font-awesome.css';

const styleSheet = createStyleSheet('SideNav', theme => ({
  listWrapper: {
    backgroundColor: theme.palette.accent[900],
    transition: theme.transitions.create(['width', 'visibility']),
    height: '100%',
    width: 190,
    borderRadius: 0,
    display: 'flex',
    flex: {
      direction: 'column',
    },
    [theme.breakpoints.up('sm')]: {
      width: 190,
    },
  },
  sideNavMini: {
    width: 0,
    visibility: 'hidden',
    [theme.breakpoints.up('sm')]: {
      visibility: 'visible',
      width: 50,
    },
  },
  label: {
    color: theme.palette.accent[200],
    flex: 1,
    textTransform: 'none',
  },
  labelMini: {
    display: 'none',
  },
  icon: {
    color: theme.palette.accent[200],
  },
  list: {
    flex: {
      grow: 1,
    },
  },
  listItem: {
    padding: 0,
    height: 48,
    borderLeftWidth: 3,
    borderLeft: 'solid transparent',
    '&:hover': {
      borderLeftWidth: 3,
      borderLeft: `solid ${theme.palette.primary[500]}`,
    },
  },
  button: {
    display: 'flex',
    justifyContent: 'center',
    width: '100%',
    height: '100%',
    minWidth: 50,
  },
  versionHide: {
    display: 'none',
  },
}));

const sideNavItems = [
  {
    link: '/log',
    icon: 'file-text-o',
    label: 'Log',
  },
  {
    link: '/execute',
    icon: 'cog',
    label: 'Execute',
  },
  {
    link: '/config',
    icon: 'pencil',
    label: 'Config',
  },
  {
    link: '/history',
    icon: 'history',
    label: 'History',
  },
  {
    link: '/movies',
    icon: 'film',
    label: 'Movies',
  },
  {
    link: '/pending',
    icon: 'check',
    label: 'Pending',
  },
  {
    link: '/schedule',
    icon: 'calendar',
    label: 'Schedule',
  },
  {
    link: '/seen',
    icon: 'eye',
    label: 'Seen',
  },
  {
    link: '/series',
    icon: 'tv',
    label: 'Series',
  },
  {
    link: '/status',
    icon: 'heartbeat',
    label: 'Status',
  },
];

class SideNav extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    sideBarOpen: PropTypes.bool.isRequired,
  };

  renderNavItems() {
    const { classes, sideBarOpen } = this.props;
    return sideNavItems.map(({ link, icon, label }) => (
      <Link to={link} key={link}>
        <ListItem className={classes.listItem}>
          <Button color="accent" className={classes.button}>
            <Icon className={classNames('fa', `fa-${icon}`, classes.icon)} />
            <p className={
              classNames(
                classes.label,
                { [classes.labelMini]: !sideBarOpen },
              )}
            >
              {label}
            </p>
          </Button>
        </ListItem>
      </Link>
    ));
  }

  render() {
    const { classes, sideBarOpen } = this.props;
    return (
      <Paper
        className={classNames(
          classes.listWrapper,
          { [classes.sideNavMini]: !sideBarOpen },
        )}
        elevation={16}
      >
        <List className={classes.list}>
          { ::this.renderNavItems() }
        </List>
        <div className={classNames({ [classes.versionHide]: !sideBarOpen })}>
          <Version />
        </div>
      </Paper>
    );
  }
}

export default withStyles(styleSheet)(SideNav);
