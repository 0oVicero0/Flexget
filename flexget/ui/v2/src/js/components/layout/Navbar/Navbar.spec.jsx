import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import IconButton from 'material-ui/IconButton';
import { Navbar } from 'components/layout/Navbar';
import { themed, router } from 'utils/tests';

function renderNavbar(toggle, logout) {
  return router(themed(<Navbar
    toggle={toggle}
    logout={logout}
    classes={{}}
  />));
}

describe('components/layout/Navbar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      renderNavbar(jest.fn(), jest.fn())
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls toggle when the hamburger button is pressed', () => {
    const toggle = jest.fn();
    const logout = jest.fn();
    const wrapper = shallow(<Navbar toggle={toggle} logout={logout} classes={{}} />);

    wrapper.find('.fa-bars').closest(IconButton).simulate('click');
    expect(toggle).toHaveBeenCalled();
  });

  it('calls handleMenuClick when the menu button is pressed', () => {
    const toggle = jest.fn();
    const logout = jest.fn();
    const target = { a: 'target' };
    const wrapper = shallow(<Navbar toggle={toggle} logout={logout} classes={{}} />);

    const button = wrapper.find('.fa-cog').closest(IconButton);

    button.simulate('click', { currentTarget: target });
    expect(wrapper.state().open).toEqual(true);
    expect(wrapper.state().anchorEl).toEqual(target);
  });
});
