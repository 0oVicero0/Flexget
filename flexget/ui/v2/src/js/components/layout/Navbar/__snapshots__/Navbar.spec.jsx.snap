// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/layout/Navbar renders correctly 1`] = `
<div
  className="MuiPaper-paper-2677686717 MuiPaper-shadow4-941451405 MuiAppBar-root-2668177540 MuiAppBar-positionFixed-2227438490 MuiAppBar-colorPrimary-15947162"
>
  <div
    className="MuiToolbar-root-3627194460 MuiToolbar-gutters-344431791"
  >
    <button
      className="MuiButtonBase-root-3477017037 MuiIconButton-root-3609702008"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-2320069359"
      >
        <span
          aria-hidden="true"
          className="material-icons MuiIcon-root-3402991258 MuiIconButton-icon-156099627 fa fa-bars"
        />
      </span>
      <span
        className="MuiTouchRipple-root-3868442396"
      />
    </button>
    <div
      className={undefined}
    />
    <a
      href="/config"
      onClick={[Function]}
    >
      <button
        aria-label="Config"
        className="MuiButtonBase-root-3477017037 MuiIconButton-root-3609702008"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-2320069359"
        >
          <span
            aria-hidden="true"
            className="material-icons MuiIcon-root-3402991258 MuiIconButton-icon-156099627 fa fa-pencil"
          />
        </span>
        <span
          className="MuiTouchRipple-root-3868442396"
        />
      </button>
    </a>
    <a
      href="/log"
      onClick={[Function]}
    >
      <button
        aria-label="Log"
        className="MuiButtonBase-root-3477017037 MuiIconButton-root-3609702008"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-2320069359"
        >
          <span
            aria-hidden="true"
            className="material-icons MuiIcon-root-3402991258 MuiIconButton-icon-156099627 fa fa-file-text-o"
          />
        </span>
        <span
          className="MuiTouchRipple-root-3868442396"
        />
      </button>
    </a>
    <button
      aria-label="Manage"
      className="MuiButtonBase-root-3477017037 MuiIconButton-root-3609702008"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-2320069359"
      >
        <span
          aria-hidden="true"
          className="material-icons MuiIcon-root-3402991258 MuiIconButton-icon-156099627 fa fa-cog"
        />
      </span>
      <span
        className="MuiTouchRipple-root-3868442396"
      />
    </button>
  </div>
</div>
`;
